File: 0657-robot-return-to-origin/0657-robot-return-to-origin.py
from collections import defaultdict
class Solution:
    def judgeCircle(self, moves: str) -> bool:
        hm = defaultdict(int)
        for direction in moves:
            hm[direction] += 1
        if hm['L'] != hm['R']:
            return False
        if hm['U'] != hm['D']:
            return False
        return True

File: 0657-robot-return-to-origin/0657-robot-return-to-origin.cpp
using namespace std;

class Solution {
public:
    bool judgeCircle(string moves) {
        int vertical = 0;
        int horizontal = 0;
        char symbol;
        for (int i = 0; i < moves.size(); i++){
            symbol = moves[i];
            if (symbol == 'U'){ 
                vertical++;
            }
            else if (symbol == 'D'){
                vertical--;
            }
            else if (symbol == 'R'){
                horizontal++;
            }
            else{
                horizontal--;
            }
        }
        if (vertical == 0 && horizontal == 0){
            return true;
        }
        return false;
    }
};